lab-integration/src/main.ts

```diff
@@ -0,0 +1,49 @@
+import { promises as fs, watch } from "fs";
+import path from "path";
+import { TestsResult } from "./types";
+
+const filePath = path.join(process.cwd(), "../test-results.json");
+
+(async () => {
+  console.info("SCRIPT IS RUNNING...");
+
+  const stringifiedDataset = await fs.readFile(filePath, "utf8");
+  const dataset = JSON.parse(stringifiedDataset) as TestsResult[];
+  let lastTestResultDate: Date = getLastTestsResulstDate(dataset);
+
+  watch(filePath, (eventType) => {
+    if (eventType === "change") {
+      const newTestsResults = dataset.filter(
+        (testResult) => convertToDate(testResult.Data) > lastTestResultDate,
+      );
+
+      if (newTestsResults.length > 0)
+        lastTestResultDate = getLastTestsResulstDate(dataset);
+
+      console.info(lastTestResultDate);
+    }
+  });
+})();
+
+function getLastTestsResulstDate(dataset: TestsResult[]): Date {
+  const orderedNewTestsResults = dataset.sort(
+    (a, b) => convertToDate(a.Data).getTime() - convertToDate(b.Data).getTime(),
+  );
+  return convertToDate(
+    orderedNewTestsResults[orderedNewTestsResults.length - 1].Data,
+  );
+}
+
+function convertToDate(date: string): Date {
```

Comment:
Instead of this function, what if we use a library like `luxon`?

It makes handling date easier.

```ts
const date = DateTime.fromFormat('27/04/2021, 08:53:31', 'dd/LL/yyyy, HH:mm:ss');
...
dateStrings.sort((a, b) => {
  const dateA = DateTime.fromFormat(a, "dd/LL/yyyy, HH:mm:ss");
  const dateB = DateTime.fromFormat(b, "dd/LL/yyyy, HH:mm:ss");
  return dateB.diff(dateA).milliseconds;
});
```